client/packages/coldchain/src/Sensor/ListView/AppBarButtons.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useRef, useState } from 'react';
+import { AppBarButtonsPortal, LoadingButton } from '@common/components';
+import { RadioIcon } from '@common/icons';
+import { useTranslation } from '@common/intl';
+import { Environment } from 'packages/config/src';
+import { useNotification } from '@common/hooks';
+
+import { useAuthContext } from 'packages/common/src/authentication/AuthContext';
+import { useQueryClient } from 'packages/common/src';
+import { useSensorApi } from '../api/hooks/utils/useSensorApi';
+
+export const AppBarButtons = () => {
+  const t = useTranslation('coldchain');
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+  const { storeId } = useAuthContext();
+  const [isUploadingFridgeTag, setIsUploadingFridgeTag] = useState(false);
+  const { success, error } = useNotification();
+  const queryClient = useQueryClient();
+  const api = useSensorApi();
+
+  const onUpload = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    let file = e?.target?.files?.[0];
+    if (!file) return;
+
+    setIsUploadingFridgeTag(true);
+
+    const formData = new FormData();
+    formData.append('file', file);
+
+    try {
+      const result = await fetch(
+        `${Environment.UPLOAD_FRIDGE_TAG}?store-id=${storeId}`,
+        {
+          method: 'POST',
+          body: formData,
+        }
+      );
+
+      if (!result.ok) {
+        throw new Error(await result.text());
+      }
+      // TODO prettify
```

Comment:
Errors and success just appears in the error and success modal, added to this issue: https://github.com/openmsupply/open-msupply/issues/2423